core/src/banking_stage/transaction_scheduler/scheduler_metrics.rs

```diff
@@ -332,3 +334,71 @@ impl SchedulerTimingMetricsInner {
         self.receive_completed_time_us = 0;
     }
 }
+
+#[derive(Default)]
+pub struct SchedulerLeaderDetectionMetrics {
+    inner: Option<SchedulerLeaderDetectionMetricsInner>,
+}
+
+pub struct SchedulerLeaderDetectionMetricsInner {
+    slot: Slot,
+    bank_detected_time: Instant,
+    bank_detected_delay_us: u64,
+}
+
+impl SchedulerLeaderDetectionMetrics {
+    pub fn update_and_maybe_report(&mut self, bank_start: Option<&BankStart>) {
+        match (&self.inner, bank_start) {
+            (None, Some(bank_start)) => self.initialize_inner(bank_start),
+            (Some(_inner), None) => self.report_and_reset(),
+            (Some(inner), Some(bank_start)) if inner.slot != bank_start.working_bank.slot() => {
```

Comment:
nit: how about defining `bank_start.working_slot()`?

Reply:
gonna leave that as follow-up, we do this in a few places iirc. Would rather create new function and replace most/all those cases